static/js/services/roompin.js

```diff
@@ -23,40 +23,95 @@
 define([
 ], function() {
 
-	return ["$window", "$q", "alertify", "translation", "safeMessage", function($window, $q, alertify, translation, safeMessage) {
+	return ["$window", "$q", "globalContext", "alertify", "translation", "safeMessage", "randomGen", "localStorage", function($window, $q, context, alertify, translation, safeMessage, randomGen, localStorage) {
 
 		var pinCache = {};
+		var getLocalStoragePINIDForRoom = function(roomName) {
+			return "room-pin-" + roomName;
+		};
+		var lockedRoomsJoinable = !!context.Cfg.LockedRoomJoinableWithPIN;
 		var roompin = {
 			get: function(roomName) {
 				var cachedPIN = pinCache[roomName];
 				return cachedPIN ? cachedPIN : null;
 			},
 			clear: function(roomName) {
 				delete pinCache[roomName];
+				localStorage.removeItem(getLocalStoragePINIDForRoom(roomName));
 				console.log("Cleared PIN for", roomName);
 			},
 			update: function(roomName, pin) {
 				if (pin) {
 					pinCache[roomName] = pin;
-					alertify.dialog.alert(translation._("PIN for room %s is now '%s'.", safeMessage(roomName), safeMessage(pin)));
+					localStorage.setItem(getLocalStoragePINIDForRoom(roomName), pin);
+					if (lockedRoomsJoinable) {
+						alertify.dialog.alert(translation._("PIN for room %s is now '%s'.", safeMessage(roomName), safeMessage(pin)));
+					}
 				} else {
 					roompin.clear(roomName);
-					alertify.dialog.alert(translation._("PIN lock has been removed from room %s.", safeMessage(roomName)));
```

Comment:
From what I understand, the old behaviour is still available if `lockedRoomsJoinable` is set? In that case we should still show the message (same as the message above).

The new behaviour shouldn't show a modal dialog to query the pin, but notifying the user that a room is locked (or no longer locked) is fine - otherwise a user doesn't know if locking/unlocking worked.